be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -313,6 +314,27 @@ Status BetaRowsetReader::next_block(vectorized::Block* block) {
     return Status::OK();
 }
 
+Status BetaRowsetReader::next_block_view(vectorized::BlockView* block_view) {
+    SCOPED_RAW_TIMER(&_stats->block_fetch_ns);
+    if (config::enable_storage_vectorization && _context->is_vec) {
+        do {
+            auto s = _iterator->next_block_view(block_view);
+            if (!s.ok()) {
+                if (s.is_end_of_file()) {
+                    return Status::OLAPInternalError(OLAP_ERR_DATA_EOF);
+                } else {
+                    LOG(WARNING) << "failed to read next block: " << s.to_string();
+                    return Status::OLAPInternalError(OLAP_ERR_ROWSET_READ_FAILED);
+                }
+            }
+        } while (block_view->empty());
+    } else {
+        return Status::NotSupported("block view only support enable_storage_vectorization");
```

Comment:
If user set `enable_storage_vectorization` to false, he/she can never execute sql until turning this to `true` and restart cluster? I think we should make this more friendly, for example, if `enable_storage_vectorization` is false, use `next_batch` to do compaction or query execution

Reply:
use ```cpp get_data_by_ref = get_data_by_ref && _rs_reader->support_return_data_by_ref() &&
                       config::enable_storage_vectorization``` to control it now.